app/main.py

```diff
@@ -174,18 +143,39 @@ def format_datum(datum):
             count=int(count)
         ) for date, lat, long, count in datum]
 
-    with conn.cursor() as cursor:
-        cursor.execute(SQL)
-        data = cursor.fetchall()
-        return PlantGrowthDataResponse(
-            plant_growth_data=[PlantIdData(species=s, plant_growth_datum=format_datum(d)) for s, d in data]
-        )
+    species_data = get_species_data(conn, center_lat, center_lon, scan_range, day_range)
+    return PlantGrowthDataResponse(
+        plant_growth_data=[PlantIdData(species=s, plant_growth_datum=format_datum(d)) for s, d in species_data]
+    )
+
+
+# Get the daily growth of each species
+@app.get("/estimate_species_gradients")
+def estimate_species_gradients():
+    return get_species_daily_growth(conn)
+
+
+# Get the pattern of species growth
+@app.get("/estimate_growth_pattern")
```

Comment:
Presumably I'll have the LLM query this endpoint and recieve dictionaries e.g. {Knotweed: CONSTANT}

Would a destructive growth (input 3 in data generation) then be one where one species is GROWING and the other is DECAYING ? non-destructive growth (input 2) being one is GROWING and the other is CONSTANT? and regular growth be both GROWING or both CONSTANT ?

Reply:
It's not exact due to non determinism but
Destructive exponential -> growing + decaying
Non destructive exponential -> growing + constant
Normal growth - > constant + constant

Note: the reverse doesn't hold, hence the need for an expert to interpret the results. 
If we find out that plant one is growing and plant two is decaying, this might not be because of destructive exponential growth but because we've planted lots of plant one recently and an animal could be eating plant two

Reply:
yeah i figured there'd be nuance in a real use case, i'm purely asking in the context of the demo data for now. we could maybe have the user send a follow up question like "what other possible explanations could there be for this behaviour?" and then add those scenarios as examples

Reply:
Thats a great idea we should definitely do that :)